build-tools/packages/build-tools/src/repoPolicyCheck/handlers/fluidBuildTasks.ts

```diff
@@ -512,6 +512,70 @@ function getTscCommandDependencies(
 	return deps.concat(defaultDeps);
 }
 
+interface BuildDepsCallbackContext {
+	packageDir: string;
+	json: PackageJson;
+	script: string;
+	command: string;
+	deps: string[];
+	root: string;
+}
+
+function buildDepsHandler(
+	file: string,
+	root: string,
+	check: (context: BuildDepsCallbackContext) => string | undefined,
+): string | undefined {
+	let json: PackageJson;
+	try {
+		json = JSON.parse(readFile(file));
+	} catch (err) {
+		return "Error parsing JSON file: " + file;
+	}
+
+	if (!isFluidBuildEnabled(root, json)) {
+		return;
+	}
+	if (json.scripts === undefined) {
+		return;
+	}
+	const packageDir = path.dirname(file);
+	const errors: string[] = [];
+	const deps = getDefaultTscTaskDependencies(root, json);
+	const ignore = getFluidBuildTasksTscIgnore(root);
+	for (const script in json.scripts) {
```

Comment:
Do you really want for...in and not for...of?

Reply:
Would `for [k, v] of Object.entries(json.scripts)` prevent needing the non null assertion?

Reply:
This code is not new / written by me. It is refactored for common use - was lines 578+.
`in` will go over properties (scripts is a [non-array] object) while `of` needs an iterable (like array).
`Object.entries` does not handle (skip) values being undefined; so, some handling or non-null assertion is needed.
(Would almost be okay if we enable [exactOptionalPropertyTypes](https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes) but that just prevents the writer from sending null, but readers still have to check.)